subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy

```diff
@@ -501,42 +507,54 @@ class BuildScriptClasspathIntegrationSpec extends AbstractIntegrationSpec implem
         file("current/build.gradle").text = subprojectSource(currentTimestampJar)
 
         expect:
-        succeeds("printMessage")
+        succeeds("printMessage", "--info")
+
+        getArtifactTransformJarsByName("original/testClasses.jar").size() == 1
+        getArtifactTransformJarsByName("instrumented/testClasses.jar").size() == 1
+    }
+
+    void notInJar9Cache(String filename) {
+        inJar9Cache(filename, false)
+    }
 
-        getCachedTransformedJarsByName("testClasses.jar").size() == 1
+    TestFile inJar9Cache(String filename, boolean shouldBeFound = true) {
+        String fullpath = result.output.readLines().find { it.matches(">>>file:.*${filename}") }.replace(">>>", "")
+        assert fullpath.startsWith(jars9CacheDir.toURI().toString()) == shouldBeFound
+        return new TestFile(new File(URI.create(fullpath)))
     }
 
-    void notInJarCache(String filename) {
-        inJarCache(filename, false)
+    TestFile notInArtifactTransformCache(String filename) {
+        inArtifactTransformCache(filename, false)
     }
 
-    TestFile inJarCache(String filename, boolean shouldBeFound = true) {
+    TestFile inArtifactTransformCache(String filename, boolean shouldBeFound = true) {
         String fullpath = result.output.readLines().find { it.matches(">>>file:.*${filename}") }.replace(">>>", "")
-        assert fullpath.startsWith(cacheDir.toURI().toString()) == shouldBeFound
+        assert fullpath.startsWith(artifactTransformCacheDir.toURI().toString()) == shouldBeFound
         return new TestFile(new File(URI.create(fullpath)))
     }
 
-    TestFile getGcFile() {
-        return cacheDir.file("gc.properties")
+
+    TestFile getJars9GcFile() {
+        return jars9CacheDir.file("gc.properties")
     }
 
-    TestFile getCacheDir() {
+    TestFile getJars9CacheDir() {
         return userHomeCacheDir.file(DefaultClasspathTransformerCacheFactory.CACHE_KEY)
     }
 
+    TestFile getArtifactTransformCacheDir() {
+        return userHomeCacheDir.file(CacheLayout.TRANSFORMS.key)
+    }
+
     /**
      * Finds all cached transformed JARs named {@code jarName}.
      * @param jarName the name of the JAR to look
      * @return the list of transformed JARs in the cache
      */
-    List<File> getCachedTransformedJarsByName(String jarName) {
-        Arrays.stream(cacheDir.listFiles()).filter {
-            File cacheChild -> isCachedTransformedEntryDir(cacheChild)
-        }.map {
-            File cacheChild -> new File(cacheChild, jarName)
-        }.filter {
-            it.exists()
-        }.collect(Collectors.toList())
+    List<File> getArtifactTransformJarsByName(String jarName) {
+        return Files.find(artifactTransformCacheDir.toPath(), 4, (path, attributes) -> normaliseFileSeparators(path.toString()).endsWith(jarName))
```

Comment:
:nail_care: could you please extract 4 into a variable to save a trip to javadoc? maybe with a comment why this depth is enough.

Reply:
I think we don't have any problem with any depth here, I changed it to Integer.MAX_VALUE